File: timerx-core/src/main/java/timerx/exceptions/IllegalSymbolsCombinationException.java
package timerx.exceptions;

// @formatter_off
/**
 * This exception is thrown when symbols in input format are in unacceptable combination,
 * such as:<br/>
 * 1) Input format has hours with seconds or milliseconds, but does not have minutes.<br/>
 * 2) Input format has hours, minutes, and milliseconds, but does not have seconds<br/>
 * 3) Input format has minutes and milliseconds, but does not have seconds
 */
public class IllegalSymbolsCombinationException extends RuntimeException {

  public IllegalSymbolsCombinationException(String message) {
    super(message);
  }
}

File: timerx-core/src/main/java/timerx/IllegalSymbolsCombinationException.kt
package timerx

/**
 * This exception is thrown when symbols in the input format are in an unacceptable combination,
 * such as:<br></br>
 * 1) Input format has hours with seconds or milliseconds, but does not have minutes.<br></br>
 * 2) Input format has hours, minutes, and milliseconds, but does not have seconds.<br></br>
 * 3) Input format has minutes and milliseconds, but does not have seconds.
 */
public class IllegalSymbolsCombinationException(message: String?) : RuntimeException(message)